src/pages/patients/Patients.tsx

```diff
@@ -61,6 +61,36 @@ const Patients = (props: PatientsProps) => {
     const indexOfFirstCard =  indexOfLastCard - itemsPerPage;
     const currentPatientList = patientList.slice(indexOfFirstCard, indexOfLastCard);
 
+    const [getPatientsByStatusMutation] = useGetPatientsByStatusMutation();
+
+    const handleGetPatientsByStatus = async () => {
+        try {
+            setIsLoading(true);
+            const statusList = Object.entries(filterStatus)
+                .filter(([key, value]) => value)
+                .map(([key]) => key);
+
+            const data = await Promise.all(
+                statusList.map(async (status) => {
+                    const result = await getPatientsByStatusMutation({ status });
+                    return result;
+            })
+            );
+            const filterData = data.flatMap((result) => 'data' in result ? result.data : []);
```

Comment:
why do you need to add this 'data'? if you cast the data from the api it doesn;t work?